### PR TITLE
R3x sdk compatibility

### PR DESCRIPTION
**Pull request by DenverM80**
Update the JavaCLI to use DS3 SDK 3.0.0-RC1

### CODE REVIEW

File: ds3-cli-integration/src/main/java/com/spectralogic/ds3cli/integration/models/JobResponse.java [hunk 1]
```diff
@@ -65,47 +67,57 @@ public JobDetails getJobDetails() {
         @JsonProperty("Nodes")
```
Comment (rpmoore) [inline ds3-cli-integration/src/main/java/com/spectralogic/ds3cli/integration/models/JobResponse.java]: Why was this field left capitalized?

File: ds3-cli-integration/src/main/java/com/spectralogic/ds3cli/integration/models/JobResponse.java [hunk 2]
```diff
@@ -65,47 +67,57 @@ public JobDetails getJobDetails() {
...
 
-        @JsonProperty("ChunkClientProcessingOrderGuarantee")
+        @JsonProperty("chunkClientProcessingOrderGuarantee")
         private String chunkClientProcessingOrderGuarantee;
 
-        @JsonProperty("Status")
+        @JsonProperty("status")
         private String status;
 
         @JsonProperty("Objects")
```
Comment (rpmoore) [inline ds3-cli-integration/src/main/java/com/spectralogic/ds3cli/integration/models/JobResponse.java]: Same question for this field.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetJobsView.java [hunk 3]
```diff
@@ -13,24 +15,24 @@
 
 public class GetJobsView implements View<GetJobsResult> {
     @Override
-    public String render(final GetJobsResult obj) throws JsonProcessingException {
+    public String render(final GetJobsResult result) throws JsonProcessingException {
 
-        final List<JobInfo> info = obj.getJobs();
-        if (info == null || info.isEmpty()) {
+        final JobsApiBean jobsInfo= result.getJobs();
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetJobsView.java]: missing space on the left side of the `=`

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/BlackPearlUtils.java [hunk 4]
```diff
@@ -19,16 +19,17 @@
...
 import java.security.SignatureException;
+import java.util.List;
 
 public final class BlackPearlUtils {
 
     public static void checkBlackPearlForTapeFailure(final Ds3Client client) throws IOException, SignatureException, CommandException {
-        final Iterable<TapeFailure> tapeFailures = client.getTapeFailure(new GetTapeFailureRequest()).getTapeFailures();
+        final List<TapeFailure> tapeFailures = client.getTapeFailuresSpectraS3(new GetTapeFailuresSpectraS3Request()).getTapeFailureListResult().getTapeFailure();
 
         if (tapeFailures != null && !Iterables.isEmpty(tapeFailures)) {
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/BlackPearlUtils.java]: now that you are using a List and not Iterable you can use a Guard:
if (Guard.isNotNullAndNotEmpty(tapeFailures))

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/SyncUtils.java [hunk 5]
```diff
@@ -49,13 +51,13 @@ public static boolean isSyncSupported(final Ds3Client client) throws IOException
     }
 
     public static boolean isNewFile(final Path localFile, final Contents serverFile, final boolean isPutCommand) throws IOException {
-        return isNewFileHelper(Files.getLastModifiedTime(localFile).toString(), serverFile.getLastModified(), isPutCommand);
+        return isNewFileHelper(Files.getLastModifiedTime(localFile).toString(), serverFile.getLastModified().toString(), isPutCommand);
     }
 
     private static boolean isNewFileHelper(final String localFileLastModifiedTime, final String serverFileLastModifiedTime, final boolean isPutCommand) {
-
+        DateTimeFormatter fmt = DateTimeFormat.forPattern("EEE MMM dd H:m:s z Y");
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/SyncUtils.java]: missing a final

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetJobsView.java [hunk 6]
```diff
@@ -13,24 +15,24 @@
...
+        return ASCIITable.getInstance().getTable(getHeaders(), fromJobInfoList(jobsInfo));
     }
 
-    private String[][] fromJobInfoList(final List<JobInfo> result) {
-        final String [][] formatArray = new String[result.size()][];
-        for(int i = 0; i < result.size(); i ++) {
-            final JobInfo job = result.get(i);
+    private String[][] fromJobInfoList(final JobsApiBean jobs) {
+        final String [][] formatArray = new String[jobs.getJobs().size()][];
+        for(int i = 0; i < jobs.getJobs().size(); i ++) {
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetJobsView.java]: missing space after the for

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapesView.java [hunk 7]
```diff
@@ -20,36 +20,32 @@
...
-    private String[][] formatTapeList(final Tapes result) {
-        final List<Tape> tapes = result.getTapes();
-        final String [][] formatArray = new String[tapes.size()][];
-        for(int i = 0; i < tapes.size(); i ++) {
-            final Tape tape = tapes.get(i);
-            final String [] bucketArray = new String[6];
-            bucketArray[0] = nullGuard(tape.getBarcode());
+    private String[][] formatTapeList(final TapeList tapeList) {
+        final String [][] formatArray = new String[tapeList.getTape().size()][];
+        for(int i = 0; i < tapeList.getTape().size(); i ++) {
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapesView.java]: missing space after for

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapesWithFullDetailsView.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        if( (result == null) || (null == result.getNamedDetailedTape()) ){
+            return "You do not have any tapes";
+        }
+
+        return ASCIITable.getInstance().getTable(getHeaders(), formatTapeList(result));
+    }
+
+    private String[][] formatTapeList(final NamedDetailedTapeList tapeList) {
+        final String [][] formatArray = new String[tapeList.getNamedDetailedTape().size()][];
+        for(int i = 0; i < tapeList.getNamedDetailedTape().size(); i ++) {
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetTapesWithFullDetailsView.java]: missing a space after for

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetTapesView.java [hunk 9]
```diff
@@ -19,15 +19,15 @@
...
 
 public class GetTapesView implements View<GetTapesResult> {
     @Override
     public String render(final GetTapesResult obj) throws JsonProcessingException {
-        final Tapes result = obj.getTapes();
+        final TapeList result = obj.getTapes();
         final CommonJsonView view = CommonJsonView.newView(CommonJsonView.Status.OK);
 
-        if( (result == null) || (null == result.getTapes()) ){
+        if( (result == null) || (null == result.getTape()) ){
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetTapesView.java]: missing space after if

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetTapesWithFullDetailsView.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+import com.spectralogic.ds3cli.util.JsonMapper;
+import com.spectralogic.ds3client.models.NamedDetailedTapeList;
+
+public class GetTapesWithFullDetailsView implements View<GetTapesWithFullDetailsResult> {
+    @Override
+    public String render(final GetTapesWithFullDetailsResult obj) throws JsonProcessingException {
+        final NamedDetailedTapeList result = obj.getTapesWithDetails();
+        final CommonJsonView view = CommonJsonView.newView(CommonJsonView.Status.OK);
+
+        if( (result == null) || (null == result.getNamedDetailedTape()) ){
```
Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetTapesWithFullDetailsView.java]: missing space after if

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetTapesWithFullDetailsView.java]: This also as extra spaces on the inside of the outside parans.

**Comment by DenverM80** [comment]
updated

**Comment by shabtaisharon** [comment]
looks good to me

**Pull request merged**